src/index.ts

```diff
@@ -25,9 +21,20 @@ const configuration = parseArgs({
 			type: "boolean",
 			default: false,
 		},
+		yarn: {
+			type: "boolean",
+			default: false,
+		},
 	},
 });
 
+const {
+	readWorkspaceDirs,
+	readWorkspaceSettings,
+	resolveWorkspaceDeps,
+	isRootDir,
+} = configuration.values.yarn ? yarn : pnpm;
```

Comment:
Package manager can be detected by traversing path and checking for `package.json/workspaces` or `pnpm-workspace.yaml`